compiler/resolution/callDestructors.cpp

```diff
@@ -739,14 +741,38 @@ void ReturnByRef::transformMove(CallExpr* moveExpr)
   callExpr->insertAtTail(tmpVar);
 
   callExpr->insertBefore(new DefExpr(tmpVar));
+  SymExpr* fnSym = toSymExpr(callExpr->baseExpr);
+  bool returnsAliasingArray = false;
+  if (fnSym && fnSym->symbol()->hasFlag(FLAG_RETURNS_ALIASING_ARRAY)) {
+    returnsAliasingArray = true;
+  }
   callExpr->insertAfter(new CallExpr(PRIM_MOVE, useLhs, tmpVar));
 
   // Possibly reduce a copy operation to a simple move
   // the copyExpr might be a copy added when normalizing initialization
   // of user variables. *or* it might come from handling `in` intent.
   if (copyExpr) {
     if (isRhsInitOrAutoCopy) {
-      removeInitOrAutoCopyPostResolution(copyExpr);
+      if (!returnsAliasingArray) {
+        removeInitOrAutoCopyPostResolution(copyExpr);
+      } else {
+        //
+        // If this call returns an aliasing array (an array view), we
+        // will need to deep-copy the result in non-'ref' situations
+        // e.g., var B = A[2..n-1]; or var B = A.reshape(2..n-1) so
+        // will need to autodestroy the result at the end of the scope
+        // and recgonize it as an array view in other situations.  Mark
+        // it as an array view to help with this in later passes.
+        //
+        SymExpr* rhsExpr = toSymExpr(copyExpr->get(1));
+        if (rhsExpr) {
+          copiesToNoDestroy = false;  // Is this too big a hammer?  redundant?
```

Comment:
Based on the other use of this flag, it seems like it cares more about whether or not the LHS is set with `FLAG_NO_AUTO_DESTROY`?

Reply:
I didn't understand this comment well enough to know what to do with it.  I'm definitely open to understanding it better and acting on it post-merge, though.

Reply:
I was musing about whether there is anything to actually do here (the redundant part of your comment).